src/neuroglancer/sliceview/renderlayer.ts

```diff
@@ -141,258 +118,27 @@ class SliceViewShaderBuffers extends RefCounted {
-          this.data.vertexIndices.subarray(frontVertexIndex * 48, (frontVertexIndex + 1) * 48));
-    }
-
-    // Compute projection matrix that transforms chunk layout coordinates to device coordinates.
-    gl.uniformMatrix4fv(
-        shader.uniform('uProjectionMatrix'), false,
-        mat4.multiply(tempMat4, dataToDeviceMatrix, chunkLayout.transform));
-
-    gl.uniform3fv(shader.uniform('uVoxelSize'), spec.voxelSize);
-    gl.uniform3fv(shader.uniform('uLowerClipBound'), spec.lowerClipBound);
-    gl.uniform3fv(shader.uniform('uUpperClipBound'), spec.upperClipBound);
-    if (DEBUG_VERTICES) {
-      (<any>window)['debug_sliceView_uVoxelSize'] = spec.voxelSize;
-      (<any>window)['debug_sliceView'] = sliceView;
-      (<any>window)['debug_sliceView_dataToDevice'] = dataToDeviceMatrix;
-    }
-  }
-
-  setupChunkDataSize(gl: GL, shader: ShaderProgram, chunkDataSize: vec3) {
-    gl.uniform3fv(shader.uniform('uChunkDataSize'), chunkDataSize);
-
-    if (DEBUG_VERTICES) {
-      (<any>window)['debug_sliceView_chunkDataSize'] = chunkDataSize;
-    }
-  }
-
-  drawChunk(gl: GL, shader: ShaderProgram, chunkPosition: vec3) {
-    gl.uniform3fv(shader.uniform('uTranslation'), chunkPosition);
-    gl.drawArrays(gl.TRIANGLE_FAN, 0, 6);
-
-    if (DEBUG_VERTICES) {
-      let sliceView: SliceView = (<any>window)['debug_sliceView'];
-      let chunkDataSize: vec3 = (<any>window)['debug_sliceView_chunkDataSize'];
-      let voxelSize: vec3 = (<any>window)['debug_sliceView_voxelSize'];
-      console.log(
-          `Drawing chunk: ${vec3Key(chunkPosition)} of data size ${vec3Key(chunkDataSize)}`);
-      let dataToDeviceMatrix: mat4 = (<any>window)['debug_sliceView_dataToDevice'];
-      this.computeVerticesDebug(
-          chunkDataSize, voxelSize, sliceView.viewportPlaneDistanceToOrigin,
-          sliceView.viewportAxes[2], chunkPosition, dataToDeviceMatrix);
-    }
-  }
-};
-
-export class RenderLayer extends GenericRenderLayer {
+export abstract class RenderLayer extends GenericRenderLayer {
   chunkManager: ChunkManager;
-  sources: VolumeChunkSource[][]|null = null;
+  sources: SliceViewChunkSource[][]|null = null;
   shader: ShaderProgram|undefined = undefined;
   shaderUpdated = true;
-  vertexComputationManager: VolumeSliceVertexComputationManager;
   rpcId: RpcId|null = null;
   shaderError: WatchableShaderError;
-  constructor(multiscaleSource: MultiscaleVolumeChunkSource, {
+  constructor(chunkManager: ChunkManager, spec: SliceViewChunkSpecification, {
     shaderError = makeWatchableShaderError(),
-    volumeSourceOptions = <VolumeSourceOptions>{}
   } = {}) {
     super();
+
     this.shaderError = shaderError;
     shaderError.value = undefined;
-    this.chunkManager = multiscaleSource.chunkManager;
+    this.chunkManager = chunkManager;
     let gl = this.gl;
-    this.vertexComputationManager = VolumeSliceVertexComputationManager.get(gl);
 
-    let sources = this.sources = multiscaleSource.getSources(volumeSourceOptions);
```

Comment:
Maybe the common code for dealing with the sources can actually remain here, rather than be duplicated?